### DISCUSSION TITLE
[SOLVED] Weblate config with Java+React project in GitHub repository

### DISCUSSION THREAD

**Discussion by joanesinurrietasmartpm** [WeblateOrg/weblate#9269] [Q&A] [answered]
I have an app with the frontend in React and backend in java.

We want to use weblate to manage the translations for the app.

When creating a component into a project in weblate, it appears one option which is to connect to a repository but I don't know if I understand the purpose of this. Is it posible to connect to the frontend and let weblate read all the source code of the project in order to get all the strings that have to be translated? (The files are mainly .jsx)

In case this can't be done, Is any way to automatize the process of getting the existing strings from a project instead of inserting them one by one?

Thank you in advance.

**Comment by nijel** [comment, marked as answer]
For Java properties the source files will typically contain just string IDs and the actual strings are defined in the properties files, so this really cannot be automated. For formats where the source file contains the actual string (such as GNU gettext), this can be automated, though not in Weblate right now (see https://github.com/WeblateOrg/weblate/issues/9189#issuecomment-1547408532).

**Reply by joanesinurrietasmartpm** [reply]
Thank you for the fast response.
